### DISCUSSION TITLE
How do I round-trip a CSV file while accessing only some of the rows?

### DISCUSSION THREAD

**Discussion by vi** [BurntSushi/rust-csv#365] [Q&A]
I want to use the convenience of `csv::Reader::deserialize` automatically mapping columns to struct fields while also preserving unknown columns found in CSV as is.

Currently I use a hack like this:

```
#[derive(Serialize,Deserialize)]
struct Entry {
    interesting_column: f64,
    another_interesting_column: f64,

    #[serde(flatten)]
    just_copy_this_from_input_to_output: BTreeMap<String, serde_json::Value>,
}
```

, but preserving other columns is tricky - I need to write csv manually, without serializing. Also using `String` instead of `serde_json::Value` makes it reject numbers in those unknown columns.

Is there some better way to handle this (without manually interpreting csv header)?

Maybe there is other deserialisation function that also gives a `csv::ByteRecord` / `csv::StringRecord` with the rest of fields? Or I can somehow `#[serde(flatten)]` the record type directly instead of resorting to a map?

**Comment by BurntSushi** [comment]
I'm on mobile or else I would write it out, but have you looked at this? https://docs.rs/csv/latest/csv/struct.ByteRecord.html#method.deserialize

I think that should be able to do what you want.

Otherwise you might need to abdicate serde.
